compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1788,7 +1788,23 @@ impl<'a> Parser<'a> {
                     }
                 }
             } else {
-                self.expected_ident_found()
+                let mut err = self.expected_ident_found();
+                if let Some((ident, _)) = self.token.ident() && ident.as_str() == "let" {
+                    self.bump(); // `let`
+                    let span = self.prev_token.span.until(self.token.span);
+                    err.span_suggestion(
```

Comment:
@Xiretza this is a good example of a `#[derive(Subdiagnostic)]` candidate that both emits notes and a suggestion!